packages/generator/src/parser/edmx-to-vdm-v2.ts

```diff
@@ -0,0 +1,140 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { ServiceNameFormatter } from '../service-name-formatter';
+import { VdmNavigationProperty, VdmComplexType, VdmEntity } from '../vdm-types';
+import {
+  EdmxAssociationSet,
+  EdmxAssociation,
+  EdmxEntityType,
+  EdmxMetadata
+} from './parser-types-v2';
+import { stripNamespace } from './parser-util';
+import {
+  JoinedEntityMetadata,
+  ParsedServiceMetadata
+} from './parser-types-common';
+import {
+  joinEntityMetadata,
+  createEntityClassNames,
+  transformEntity,
+  navigationPropertyBase
+} from './edmx-to-vdm-common';
+
+export function joinAssociationMetadata(
+  associationSets: EdmxAssociationSet[],
+  associations: EdmxAssociation[]
+): JoinedAssociationMetadata[] {
+  return associationSets.map(assocSet => {
+    const matchingAssoc = associations.find(
+      a => a.Name === stripNamespace(assocSet.Association)
+    );
+
+    if (!matchingAssoc) {
```

Comment:
Why this can never or should happen? Can you explain what is the content of the associationSets and the associations.

Reply:
This is the old v2 behavior. If the edmx file is corrupt is it possible that there is no corresponding association found.

Reply:
ok